pennylane/utils.py

```diff
@@ -353,3 +356,50 @@ def __str__(self):
                 output += "{}({}(wires={}))\n".format(return_map[op.return_type], op.name, op.wires)
 
         return output
+
+
+def inv(operation_list):
+    """Invert a list of operations or a template.
+
+    If the inversion happens inside a QNode, the operations are removed and requeued
+    in the reversed order for proper inversion.
+
+    Args:
+        operation_list (Iterable[~.Operation]): An iterable of operations
+
+    Returns:
+        List[~.Operation]: The inverted list of operations
+    """
+    if isinstance(operation_list, qml.operation.Operation):
+        operation_list = [operation_list]
+    elif not isinstance(operation_list, Iterable):
+        raise ValueError("The given operation_list is not iterable.")
+
+    non_ops = [
+        (op, idx)
```

Comment:
Small subtlety: the order of ``op`` and  ``idx`` seems to be reversed here compared to how you are unpacking them from ``enumerate(operation_list)``. Then on line 384 in ``string_reps = [" operation_list[{}] = {}".format(idx, op) for op, idx in non_ops]`` they seem to be reversed again to correspond to the original order.

If I understand it correctly, the ordering could be reserved throughout these operations, right?

Reply:
No, this will not be an issue, it's just a matter of taste. But I agree that I should stick with the usual python pattern if `(idx, obj)`. I'll change it!